src/components/IGVFileSelector.js

```diff
@@ -13,27 +12,80 @@ const getStrings = (v) => {
   return Utils.cond([_.isString(v), () => [v]], [!!v?.items, () => _.flatMap(getStrings, v.items)], () => []);
 };
 
-export const getValidIgvFiles = (values) => {
-  return _.flatMap((value) => {
-    const possibleFile = /(.+)\.([^.]+)$/.exec(value);
+const splitExtension = (fileUrl) => {
+  const extensionDelimiterIndex = fileUrl.lastIndexOf('.');
+  const base = fileUrl.slice(0, extensionDelimiterIndex);
+  const extension = fileUrl.slice(extensionDelimiterIndex + 1);
+  return [base, extension];
+};
 
-    if (_.isEmpty(parseGsUri(value)) || !possibleFile) {
-      return [];
-    }
+const isTdrUrl = (fileUrl) => {
+  const parts = fileUrl.split('/');
+  return (
+    parts.length === 6 &&
+    /datarepo-[a-f0-9]+-bucket/.test(parts[2]) &&
```

Comment:
Can we make this check work for non-prod environments?

`datarepo` is the prefix for TDR dataset bucket names in production, but other environments have a prefix of the form `datarepo-<env>`:
  - Alpha -> `datarepo-alpha`
  - Dev -> `datarepo-dev`
  - Perf -> `datarepo-perf`
  - Staging -> `datarepo-staging`
  - Tools -> `datarepo-tools`

Some additional context around bucket naming: 2 years ago we started delegating Google project creation to Resource Buffer Service (RBS), but there are a small handful of older datasets with bucket names that don't follow our RBS-era naming convention.  That being said, I don't think it's worth making this backwards-compatible: most of these buckets are for datasets that are no longer in active use, or are used for testing… if their users wanted to use IGV to look at their genomics files, we would have heard about it by now :)

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4084/commits/172d7758183b8dfef10a450fc0b19ff7c85c2a8d.